examples/gps/gps_example.cc

```diff
@@ -0,0 +1,202 @@
+    cachebasedir = osm_gps_map_get_default_cache_directory();
+
+    if (opt_cache_base_dir && g_file_test(opt_cache_base_dir, G_FILE_TEST_IS_DIR)) {
+        cachedir = g_strdup(OSM_GPS_MAP_CACHE_AUTO);
+        cachebasedir = g_strdup(opt_cache_base_dir);
+    } else if (opt_friendly_cache) {
+        cachedir = g_strdup(OSM_GPS_MAP_CACHE_FRIENDLY);
+    } else if (opt_no_cache) {
+        cachedir = g_strdup(OSM_GPS_MAP_CACHE_DISABLED);
+    } else {
+        cachedir = g_strdup(OSM_GPS_MAP_CACHE_AUTO);
+    }
+    
+    map = (OsmGpsMap*) g_object_new (OSM_TYPE_GPS_MAP,
+                        "map-source",opt_map_provider,
+                        "tile-cache",cachedir,
+                        "tile-cache-base", cachebasedir,
+                        "proxy-uri",g_getenv("http_proxy"),
+                        "drag-limit",1,
+                        "auto-download", TRUE,
+                        "gps-track-point-radius", 1,
+                        NULL);
+
+    osd = (OsmGpsMapLayer*) g_object_new (OSM_TYPE_GPS_MAP_OSD,
+                        "show-scale",TRUE,
+                        "show-coordinates",TRUE,
+                        "show-crosshair",TRUE,
+                        "show-dpad",TRUE,
+                        "show-zoom",TRUE,
+                        "show-gps-in-dpad",TRUE,
+                        "show-gps-in-zoom",TRUE,
+                        "dpad-radius", 30,
+                        NULL);
+
+    
+    
+
+    
+
+
+    osm_gps_map_layer_add(OSM_GPS_MAP(map), osd);
+    g_object_unref(G_OBJECT(osd));
+
+    builder = gtk_builder_new();
+
+    //Enable keyboard navigation
+    osm_gps_map_set_keyboard_shortcut(map, OSM_GPS_MAP_KEY_FULLSCREEN, GDK_KEY_F11);
+    osm_gps_map_set_keyboard_shortcut(map, OSM_GPS_MAP_KEY_UP, GDK_KEY_Up);
+    osm_gps_map_set_keyboard_shortcut(map, OSM_GPS_MAP_KEY_DOWN, GDK_KEY_Down);
+    osm_gps_map_set_keyboard_shortcut(map, OSM_GPS_MAP_KEY_LEFT, GDK_KEY_Left);
+    osm_gps_map_set_keyboard_shortcut(map, OSM_GPS_MAP_KEY_RIGHT, GDK_KEY_Right);
+
+    //Build the UI  
+    builder = gtk_builder_new();
+    gtk_builder_add_from_file (builder, "mapviewer.ui", &error);
+    if (error)
+        g_error ("ERROR: %s\n", error->message);
+
+    gtk_box_pack_start (
+                GTK_BOX(gtk_builder_get_object(builder, "map_box")),
+                GTK_WIDGET(map), TRUE, TRUE, 0);
+
+    widget = GTK_WIDGET(gtk_builder_get_object(builder, "window1"));
+    gtk_widget_show_all (widget);
+
+    Gps gps;
+    units::angle::degree_t lat;
+    units::angle::degree_t lon;
+    units::length::meter_t alt;  
+
+    gps.getPosition(lat,lon, alt);
+
+    osm_gps_map_set_center_and_zoom (map,
+                                     50.8636,
+                                     -0.096366,
+                                     70);
+
+    // add updater function
+
+    MapAndGps *mgps = new MapAndGps(map, &gps);
+    g_timeout_add (1000, on_timeout, mgps);
+    gtk_main();
+    delete mgps;
```

Comment:
You don't need this ``new`` and ``delete`` here.

Just do:
```c++
    MapAndGps mgps{ map, &gps };
    g_timeout_add (1000, on_timeout, &mgps);
    gtk_main();
```

Reply:
Also in general you want to avoid using ``new`` and ``delete`` altogether in modern C++. Your alternatives are:

1. Allocate on the stack rather than the heap (like here)
2. If you *have* to allocate an array or something on the heap, use a wrapper object like a ``std::vector``
3. If you *need* a pointer to something that has to be allocated on the heap, use a smart pointer like a ``std::unique_ptr`` or something

Reply:
this was just a hacky way to get the example to work to test, but I agree